Python/pytime.c

```diff
@@ -1405,14 +1396,48 @@ _PyTime_gmtime(time_t t, struct tm *tm)
 _PyTime_t
 _PyDeadline_Init(_PyTime_t timeout)
 {
-    _PyTime_t now = _PyTime_GetMonotonicClock();
+    _PyTime_t now;
+    if (PyTime_Monotonic(&now) < 0) {
+        PyErr_Clear();
+        now = 0;
+    }
     return _PyTime_Add(now, timeout);
 }
 
 
 _PyTime_t
 _PyDeadline_Get(_PyTime_t deadline)
 {
-    _PyTime_t now = _PyTime_GetMonotonicClock();
+    _PyTime_t now;
+    if (PyTime_Monotonic(&now) < 0) {
+        PyErr_Clear();
+        now = 0;
+    }
     return deadline - now;
 }
+
+// Internal wrappers that ignore all excetpions.
+// (The exceptions are rare events; we don't care about performance
+// if they're raised.)
+#define NOEXC_WRAPPER(NAME, FUNC)                   \
+    PyTime_t                                        \
+    NAME(void) {                                    \
+        PyTime_t result;                            \
+        if (FUNC(&result) < 0) {                    \
```

Comment:
These functions can be called without holding the GIL. I don't this pattern is safe in that case.

Reply:
I suggest to remove this macro and keep the 3 existing internal functions which pass `0` to `raise_exc`.